packages/web-components/src/search-widget/server/utils/map-minimal-properties.ts

```diff
@@ -0,0 +1,34 @@
+import pickBy from 'lodash.pickby'
+import { PagedResultPropertyModel_, PagedResultPropertyImageModel_ } from '@reapit/foundations-ts-definitions'
+import {
+  PickedPagedResultPropertyModel_,
+  PickedPagedResultPropertyImageModel_,
+  PickedPropertyModel,
+  PickedPropertyImageModel,
+} from '../../types'
+
+type InputObject = PagedResultPropertyModel_ | PagedResultPropertyImageModel_
+type PickedModel = PickedPropertyModel | PickedPropertyImageModel
+type OutputObject = PickedPagedResultPropertyModel_ | PickedPagedResultPropertyImageModel_
+
+/**
+ * Map through to return only necessary properties
+ */
+export const mapMinimalProperties = (
+  { _embedded, ...rest }: InputObject,
+  includedEmbeddedProps: string[],
+): OutputObject => {
```

Comment:
Rather than this union type, can mapMinimalProperties not accept a generic eg `const mapMinimalProperties = <T, R>(...` where T is the input and R is the result type?

Reply:
Not a big deal, just a thought

Reply:
That's great, fixed 👍